### ISSUE TITLE
Client IP get refreshes only on logout

### ISSUE THREAD

**Issue by hc4** [Graylog2/graylog2-server#2656]
I have SSO enabled and graylog behind reverse proxy.
The problem could be tested on Users page.

1. Open Users page - correct IP displayed for current user
2. Remove all trusted_proxies from graylog config and restart graylog
3. Users page still shows old user IP (but should be proxy ip)
4. Click "Log out"
5. User logs in automaticaly (thanks to SSO plugin)
6. Now proxy IP displayed as client ip for relogged user.
7. Return trsuted proxies to config and restart graylog.
8. Again correct client IP will be displayed only after relogging

## Your Environment
* Graylog Version: 2.1.0-beta3 snapshot

**Comment by kroepke** [comment]
Weird, apparently not all session attributes are being updated when the session is validated. I'll investigate.

**Comment by hc4** [comment]
Also strange thinks happens with last activity.
I have user, for which last activity was displayed "4 months ago".
Then he logged in using SSO, and last activity became "few seconds ago".
But after he closed browser, last activity was again "4 months ago"

**Comment by kroepke** [comment]
@hc4 A user can have multiple sessions, for display purposes we get the latest one.
The user is probably configured so that her sessions never time out.
Also note that old sessions do not have their host stored with them, so previously existing sessions won't show the client IP.

**Comment by hc4** [comment]
I've changed option session never expires, but this 4 months ago session still there :)
Also I think session should be validated against client ip.
i.e. You shouldn't be able to connect from another host with same session id.

And how can I clear all user's sessions?

**Comment by hc4** [comment]
http://graylog/system/sessions shows only my session (despite I'm admin)
Can I see all sessions on server using REST API?

**Comment by kroepke** [comment]
That is currently not implemented, no. It was meant to be in #2551 but that didn't make it into 2.1

**Comment by kroepke** [comment]
Closer inspection of the framework code showed that this use case is not supported directly.
The session only saves the host where it was *originally initiated from*, not the current one. I suppose the intention was to be able to find sessions that were created from a different source than is currently being used.

We would need to save the current host differently, but that is out of scope for 2.1.

I'll leave it open so we can integrate it with #2551 and instead change the wording to make it clear that this is not the current host.
